python-package/lightgbm/basic.py

```diff
@@ -1802,9 +1802,9 @@ def _lazy_init(
 
         # process for args
         params = {} if params is None else params
-        args_names = (getattr(self.__class__, '_lazy_init')
+        args_names = (self.__class__._lazy_init
                       .__code__
-                      .co_varnames[:getattr(self.__class__, '_lazy_init').__code__.co_argcount])
+                      .co_varnames[:self.__class__._lazy_init.__code__.co_argcount])
```

Comment:
I think this may be a good opportunity to do this using the [inspect module](https://docs.python.org/3/library/inspect.html#introspecting-callables-with-the-signature-object) instead. I believe it would be something like:
`args_names = inspect.signature(self.__class__._lazy_init).parameters.keys()`. WDYT?

Reply:
I agree!

I think the code in this part of the diff is probably a holdover from when LightGBM supported Python 2 ([`inspect.signature()` was added in Python 3.3](https://docs.python.org/3/library/inspect.html#introspecting-callables-with-the-signature-object))

Just pushed https://github.com/microsoft/LightGBM/pull/5871/commits/7c64939ea6f8c406f0649c69f12a17dffe8ec42f adding this.